### PR TITLE
Example of using async commit

### PR DESCRIPTION
**Pull request by tohar-ayash**
Signed-off-by: tohar-ayash <toharayash1@gmail.com>

### CODE REVIEW

File: examples/api/async_commit/async_commit.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+		fmt.Printf("Commit failed, reason: %s\n", err.Error())
+		return
+	}
+	l, err := session.Ledger()
+	if err != nil {
+		fmt.Printf(err.Error())
+		return
+	}
+	fmt.Println("Getting transaction receipt")
+	for txReceipt == nil {
```
Comment (gennadylaventman) [inline examples/api/async_commit/async_commit.go:83]: Can you add some timeout mechanism?

File: examples/api/async_commit/async_commit.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+		UserConfig:   c.SessionConfig.UserConfig,
+		TxTimeout:    c.SessionConfig.TxTimeout,
+		QueryTimeout: c.SessionConfig.QueryTimeout}
+
+	fmt.Println("Opening session to database, configuration: ", c.SessionConfig)
+	session, err := db.Session(sessionConf)
+	if err != nil {
+		fmt.Printf("Database session creating failed, reason: %s\n", err.Error())
+		return
+	}
```
Comment (gennadylaventman) [inline examples/api/async_commit/async_commit.go:55]: Can you move all data preparation code to separate function, like you did in pr 51

File: examples/api/async_commit/async_commit.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+	fmt.Println("Adding key, value: key1, val1 to the database")
+	err = tx.Put("bdb", "key1", []byte("val1"), nil)
+	if err != nil {
+		fmt.Printf("Adding new key to database failed, reason: %s\n", err.Error())
+		return
+	}
+
+	fmt.Println("Committing transaction")
+	txID, txReceipt, err := tx.Commit(false)
```
Comment (tock-ibm) [inline examples/api/async_commit/async_commit.go:72]: add a comment and a printout that the receipt of an async commit is always `nil`

File: examples/api/async_commit/async_commit.go [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+	}
+	l, err := session.Ledger()
+	if err != nil {
+		fmt.Printf(err.Error())
+		return
+	}
+	fmt.Println("Getting transaction receipt")
+	for txReceipt == nil {
+		time.Sleep(200 * time.Millisecond)
+		txReceipt, _ = l.GetTransactionReceipt(txID)
```
Comment (tock-ibm) [inline examples/api/async_commit/async_commit.go:85]: do not ignore the error

File: examples/api/async_commit/async_commit.go [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+	}
+	fmt.Printf("The transaction is stored on block header number %d, index %d\n", txReceipt.Header.BaseHeader.Number, txReceipt.TxIndex)
```
Comment (tock-ibm) [inline examples/api/async_commit/async_commit.go:87]: check validity, print the receipt (also) using `%+v`

File: examples/api/async_commit/async_commit.go [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+	fmt.Printf("Transaction receipt = %s\n", txReceipt)
+
+	l, err := session.Ledger()
+	if err != nil {
+		fmt.Printf(err.Error())
+		return
+	}
+
+	fmt.Println("Getting transaction receipt")
+	i := 0
```
Comment (tock-ibm) [inline examples/api/async_commit/async_commit.go:52]: why do we need this `i`?

File: examples/api/async_commit/async_commit.go [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+	for start := time.Now(); time.Since(start) < time.Second; {
+		time.Sleep(200 * time.Millisecond)
+		txReceipt, err = l.GetTransactionReceipt(txID)
+		if err != nil{
+			fmt.Printf("Getting transaction receipt failed, reason: %s\n", err.Error())
+			return
+		}
+		i++
```
Comment (tock-ibm) [inline examples/api/async_commit/async_commit.go:60]: why do we need this `i`?

File: examples/api/async_commit/async_commit.go [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+	l, err := session.Ledger()
+	if err != nil {
+		fmt.Printf(err.Error())
+		return
+	}
+
+	fmt.Println("Getting transaction receipt")
+	i := 0
+	for start := time.Now(); time.Since(start) < time.Second; {
+		time.Sleep(200 * time.Millisecond)
```
Comment (tock-ibm) [inline examples/api/async_commit/async_commit.go:54]: This loop spends a second of polling... once the receipt is not nil you should break it...
There is also an alternative way to do it using `select`:
```
LOOP:
	for {
		timeout := time.After(5 * time.Second)
		select {
		case <-time.After(10 * time.Millisecond):
			//poll, if nil continue, if not nil break out of for-loop
			continue
		case <-timeout:
			//break for-loop (or return)
			break LOOP
		}
	}
```

File: examples/api/async_commit/async_commit.go [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+		return
+	}
+
+	fmt.Println("Getting transaction receipt")
+	LOOP:
+		for {
+			timeout := time.After(5 * time.Second)
+			select {
+			case <-time.After(10 * time.Millisecond):
+				time.Sleep(200 * time.Millisecond)
```
Comment (tock-ibm) [inline examples/api/async_commit/async_commit.go:57]: You don't need that... select will block until one of the cases can execute.
https://tour.golang.org/concurrency/5

File: examples/api/async_commit/async_commit.go [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+				if txReceipt == nil {
+					continue
+				} else {
+					break LOOP
+				}
+			case <-timeout:
+				fmt.Println("Getting transaction receipt failed")
+				return
+			}
+		}
```
Comment (tock-ibm) [inline examples/api/async_commit/async_commit.go:71]: do `gofmt`, this is not indented correctly

**Review by tock-ibm — APPROVED** [review]

**Pull request merged**